src/components/organisms/InformationLeftColumn/InformationLeftColumn.tsx

```diff
@@ -1,83 +1,101 @@
-import React from "react";
-import "./InformationLeftColumn.scss";
+import React, {
+  forwardRef,
+  useCallback,
+  useImperativeHandle,
+  useState,
+} from "react";
+import classNames from "classnames";
+
+import { IconDefinition } from "@fortawesome/fontawesome-common-types";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import {
   faAmbulance,
   faAngleDoubleRight,
   faHeart,
   faEdit,
+  faInfoCircle,
 } from "@fortawesome/free-solid-svg-icons";
 
-interface PropsType {
-  venueLogoPath: string;
+import "./InformationLeftColumn.scss";
+
+export type ValidLogoIconName = "ambulance" | "create" | "heart" | "info";
+
+const logoMap = new Map<ValidLogoIconName, IconDefinition>([
+  ["ambulance", faAmbulance],
+  ["create", faEdit],
+  ["heart", faHeart],
+  ["info", faInfoCircle],
+]);
+
+interface InformationLeftColumnProps {
+  iconNameOrPath?: ValidLogoIconName | string;
   children: React.ReactNode;
-  isLeftColumnExpanded: boolean;
-  setIsLeftColumnExpanded: React.Dispatch<React.SetStateAction<boolean>>;
 }
 
-const InformationLeftColumn: React.FunctionComponent<PropsType> = ({
-  venueLogoPath,
-  children,
-  isLeftColumnExpanded,
-  setIsLeftColumnExpanded,
-}) => {
+export interface InformationLeftColumnControls {
+  isExpanded: boolean;
+  setExpanded: (isExpanded: boolean) => void;
+  toggleExpanded: (e?: React.MouseEvent<HTMLElement>) => void;
+}
+
+export const InformationLeftColumn = forwardRef<
+  InformationLeftColumnControls,
+  InformationLeftColumnProps
+>(({ iconNameOrPath = "info", children }, controlsRef) => {
+  const [isExpanded, setExpanded] = useState(false);
+
+  const toggleExpanded = useCallback((e?: React.MouseEvent<HTMLElement>) => {
+    e && e.stopPropagation();
+
+    setExpanded((prev) => !prev);
+  }, []);
+
+  // Expose internal state/controls to parent components via ref
+  useImperativeHandle(controlsRef, () => ({
+    isExpanded,
+    setExpanded,
+    toggleExpanded,
+  }));
```

Comment:
FYI: This pattern allows a direct parent to gain access to 'internal state/controls' as defined/provided by a component.

- https://reactjs.org/docs/hooks-reference.html#useimperativehandle

Reply:
I feel a bit controversial about this pattern. The example shown in the React lib seems to be a very specific use-case when you can't do it any other way.

> As always, imperative code using refs should be avoided in most cases

Your case seems to be quite straightforward on the other hand. I would probably think twice before introducing patterns like this into our codebase. One way to have it implemented would be to store the UI states in the redux? It can sound like overkill, but this way we can ensure we have consistency.

Reply:
- https://levelup.gitconnected.com/reacts-useimperativehandle-made-simple-81035a21eef0
  - > The general React style is to have a unidirectional flow of data and logic; That is to say, you should pass functions and data down through props and a component should only ever be able to access what is passed in as props. In cases where bidirectional or lateral dataflow is needed, we can use libraries such as Redux or React’s context API.
  - > **However, in some cases, importing redux or using context is simply just overkill — This is where useImperativeHandle comes in.** This will provide us a lightweight solution to have bidirectional flow.
  - > It’s not something you’ll use often but in rare cases can be incredibly useful and prevent you from having to overreach for extreme solutions like redux or rack your brain coming up with solutions like passing in multiple props, stateUpdaters and isolated error messages — the later being the hardest.
  - The example they describe here is basically equivalent to how I have used it.
  
---

I don't believe it's an inherently bad pattern to have/use selectively within our codebase. It's definitely not the right answer for every situation (for example if you needed to be able to trigger this functionality from somewhere outside of the direct parent, it wouldn't make sense to use this pattern)

If we need access to it from multiple places, then would generally either use React's Context, or put the state in Redux. Particularly with our current redux patterns, the boilerplate of doing that is a huge amount of overhead, for IMO very little good reason. Moving things to global state that can exist more than once at the same time also necessitates a strategy for splitting those up, so that their global state doesn't clash, which adds additional boilerplate again.

I believe among the usages I saw, there was only 1 place it was used where we even need the parent to be able to 'get' the controls that we are exposing. My usage here is sort of in a similar yet different way to the 'functions as children' pattern (which I believe tends to have it's own performance/usage challenges, that from what I remember of working with it in the past, are more annoying than the API/usage i've got here):

- https://codedaily.io/tutorials/6/Using-Functions-as-Children-and-Render-Props-in-React-Components

The alternative to this (and the way it was before I changed it), is that every single place that wants to use this component, also needs to set up/manage the state for it, just so that it can open/close/etc; and then pass the state setter/getters into the component as props; which is something that can be seen in quite a few places across our codebase at the moment (and even more places in the plast), and it adds needless complexity and boilerplate to handle what is essentially an 'implementation detail' of the component. Solving it once within 

---

**tl;dr** of the above, I don't believe it's an inherently bad pattern, though it is one that we should consider and weigh up if/when we think it would be good to use in other places. I definitely don't think it's a bad enough pattern to block getting this user-facing bugfix/cleanup merged, that has been outstanding since at least December last year, and had been requested to be fixed multiple times since then (including most recently yesterday)